### PR TITLE
Use ESMF 8.2.0 library (feature/esmf_8_2_0)

### LINKED ISSUE #621: Use ESMF 8.2.0 library
Update the ESMF library from 8.1.0 to 8.2.0. This change is needed to utilize latest fixes in the ESMF; see, for example, #143.

A corresponding PR will be made, in which regression (consistency) tests for those utilities affected by the ESMF library (chgres_cube and grid_gen) on all 5 platforms will be run and comparison of results be posted. 

Part of #143.

### PR DESCRIPTION
**Pull request by MinsukJi-NOAA**
## DESCRIPTION OF CHANGES: 
Use ESMF 8.2.0 library. Motivation for this change comes from #143.

## TESTS CONDUCTED: 
Consistency tests have been run on all 5 supported platforms for those utilities that are affected by the ESMF library: `chgres_cube` and `grid_gen`. The results are reported [here](https://github.com/ufs-community/UFS_UTILS/issues/621)

## DEPENDENCIES:
None

## DOCUMENTATION:
N/A

## ISSUE: 
Fixes #621
Part of #143.

### CODE REVIEW

File: modulefiles/build.wcoss_dell_p3.intel.lua [hunk 1]
```diff
@@ -34,16 +34,16 @@ load(pathJoin("hdf5", hdf5_ver))
...
 
-esmf_ver=os.getenv("esmf_ver") or "8_1_0_beta_snapshot_27"
+esmf_ver=os.getenv("esmf_ver") or "8_2_0"
 load(pathJoin("esmf", esmf_ver))
 
 bacio_ver=os.getenv("bacio_ver") or "2.4.1"
 load(pathJoin("bacio", bacio_ver))
 
-g2_ver=os.getenv("g2_ver") or "3.4.1"
+g2_ver=os.getenv("g2_ver") or "3.4.2"
```
Comment (GeorgeGayno-NOAA) [inline modulefiles/build.wcoss_dell_p3.intel.lua:46]: Curious about why you upgraded the G2 version.

File: modulefiles/build.wcoss_cray.intel [hunk 2]
```diff
@@ -31,9 +31,9 @@ setenv ZLIB_ROOT /usrx/local/prod/zlib/1.2.7/intel/haswell
 setenv PNG_ROOT /usrx/local/prod/png/1.2.49/intel/haswell
 setenv Jasper_ROOT /usrx/local/prod/jasper/1.900.1/intel/haswell
 
-module use /gpfs/hps3/emc/nems/noscrub/emc.nemspara/soft/modulefiles
-#module load esmf/8.0.0
-setenv ESMFMKFILE /gpfs/hps3/emc/nems/noscrub/emc.nemspara/soft/esmf/8.0.0/lib/esmf.mk
+#module use /gpfs/hps3/emc/nems/noscrub/emc.nemspara/soft/modulefiles
```
Comment (GeorgeGayno-NOAA) [inline modulefiles/build.wcoss_cray.intel]: Instead of commenting out these lines, you can just remove them.

**Review by GeorgeGayno-NOAA — CHANGES_REQUESTED** [review]
We should update the workflow files to use ESMF 8.2.0. I can help with that.

**Comment by MinsukJi-NOAA** [comment]
> We should update the workflow files to use ESMF 8.2.0. I can help with that.

I just committed these changes in 62522c8.

File: modulefiles/build.wcoss_dell_p3.intel.lua (hunk 1 above)
Comment (MinsukJi-NOAA) [reply modulefiles/build.wcoss_dell_p3.intel.lua:46]: In order to use `esmf_8_2_0`, I had to change `ips_ver` to `18.0.5.274`, which in return required changing `g2` to `3.4.2` and  `nccmp` to `1.8.9.0` to be able to module load.

File: modulefiles/build.wcoss_cray.intel (hunk 2 above)
Comment (MinsukJi-NOAA) [reply modulefiles/build.wcoss_cray.intel]: These lines have been removed.

File: modulefiles/build.wcoss_dell_p3.intel.lua (hunk 1 above)
Comment (GeorgeGayno-NOAA) [reply modulefiles/build.wcoss_dell_p3.intel.lua:46]: Ok.

**Comment by GeorgeGayno-NOAA** [comment]
@MinsukJi-NOAA I see some of the checks failed because of memory leaks. I don't think you did anything wrong. Let me investigate what is going on.

**Comment by MinsukJi-NOAA** [comment]
> @MinsukJi-NOAA I see some of the checks failed because of memory leaks. I don't think you did anything wrong. Let me investigate what is going on.

Thanks!

**Comment by GeorgeGayno-NOAA** [comment]
> @MinsukJi-NOAA I see some of the checks failed because of memory leaks. I don't think you did anything wrong. Let me investigate what is going on.

I am able to suppress the memory links. I can update your branch.

**Comment by GeorgeGayno-NOAA** [comment]
@kgerheiser By upgrading to ESMF v8.2.0, we are getting failed tests due to memory leaks. There were no ufs_utils code changes, so I am blaming the ESMF library. Do you agree? To get the tests to pass, I have to keep adding items to the suppression file: https://github.com/GeorgeGayno-NOAA/UFS_UTILS/commit/04b7a0d94b008695c7ab825755261161c4963afd Is there a better way?

**Comment by kgerheiser** [comment]
I would agree, it's probably ESMF. Memory check flags are temperamental and it's easy to encounter situations like this. It's either keep adding to the suppression file or turn them off.

**Review by GeorgeGayno-NOAA — APPROVED** [review]

**Pull request merged**